client/src/main/java/io/pravega/client/tables/KeyValueTableMap.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import lombok.NonNull;
+
+/**
+ * {@link Map} implementation for a {@link KeyValueTable}'s Key Family.
+ * <p>
+ * All methods inherited from {@link Map} respect the contracts defined in that interface, except as noted below.
+ * <p><p>
+ * Performance considerations:
+ * <ul>
+ * <li> The {@link Map} interface defines synchronous operations, however {@link KeyValueTable} defines async operations.
+ * All method implementations defined in this interface or inherited from {@link Map} invoke
+ * {@link CompletableFuture#join()} on any {@link KeyValueTable} methods invoked. This means that two threads are used
+ * to fulfill these calls: one from the Executor passed in to the {@link KeyValueTable} factory, and one on which this
+ * request is executing.
+ *
+ * <li> The following operations result in a single call to the wrapped {@link KeyValueTable}:
+ * <ul>
+ * <li> {@link #containsKey}.
+ * <li> {@link #get}, {@link #getOrDefault}.
+ * <li> {@link #putDirect}, {@link #putAll}, {@link #putIfAbsent}.
+ * <li> {@link #removeDirect}.
+ * <li> {@link #isEmpty()} (Invokes {@link KeyValueTable#keyIterator} and requests a single item).
+ * <li> {@link #keySet()}: {@link Set#removeAll}, {@link Set#contains}, {@link Set#containsAll}, {@link Set#isEmpty()}.
+ * <li> {@link #entrySet()}: {@link Set#contains}, {@link Set#containsAll}, {@link Set#isEmpty()}, {@link Set#add}, {@link Set#addAll}.
+ * </ul>
+ *
+ * <li> The following operations result up to two calls to the wrapped {@link KeyValueTable}:
+ * <ul>
+ * <li> {@link #put} (refer to {{@link #putDirect} if the old value is not needed).
+ * <li> {@link #replace}.
+ * <li> {@link #remove} (refer to {@link #removeDirect} if the old value is not needed).
+ * <li> {@link #computeIfPresent}, {@link #compute}}, {@link #computeIfAbsent}.
+ * <li> {@link #keySet()}{@link Set#remove}.
+ * <li> {@link #entrySet()}: {@link Set#removeAll}, {@link Set#isEmpty()}.
```

Comment:
Yes, there is a difference. Calling `entrySet` by itself does nothing. But the methods listed after this (see the colon `:`) do. The reason this is listed multiple times is because I'm grouping the actual method calls by their behavior.